apiserver/facades/client/client/status.go

```diff
@@ -956,6 +967,16 @@ func (context *statusContext) processApplication(application *state.Application)
 		processedStatus.WorkloadVersion = versions[0].Message
 	}
 
+	bindings := make(map[string]string)
+	for _, bind := range context.allAppsUnitsCharmBindings.endpointBindings {
```

Comment:
Up in fetchAllApplicationsAndUnits(), we should create a map[string]map[string]string 
ie appName -> map[string]string
similar to the unitsMap so that by the time we get to here, it is a simple map look up on appName and not a for loop each time